mobilecoind/strategies/drain-accounts.py

```diff
@@ -0,0 +1,132 @@
+This is the case for minted tokens (which fog distro cannot transfer).
+
+It zips the source accounts and dest accounts together, and each source account sends
+its entire balance (less fee) to the dest account
+
+Example setup and usage:
+```
+    python3 drain-accounts.py --key-dir ../../../target/sample_data/master/keys/ --dest-key-dir ../../../target/sample_data/master/fog_keys/ --fee 20
+```
+"""
+import argparse
+import grpc
+import mobilecoind_api_pb2
+import mobilecoind_api_pb2_grpc
+import os
+import time
+from accounts import wait_for_accounts_sync
+from google.protobuf.empty_pb2 import Empty
+
+
+def parse_args() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--mobilecoind-host",
+                        default="localhost",
+                        type=str,
+                        help="Mobilecoind host")
+    parser.add_argument("--mobilecoind-port",
+                        default="4444",
+                        type=str,
+                        help="Mobilecoind port")
+    parser.add_argument("--key-dir",
+                        type=str,
+                        help="Path to directory of account_keys")
+    parser.add_argument("--dest-key-dir",
+                        type=str,
+                        help="Path to directory of account_keys")
+    parser.add_argument("--max-seconds",
+                        type=int,
+                        default=40,
+                        help="Number of seconds to wait for a tx to clearn")
+    parser.add_argument("--fee",
+                        type=int,
+                        default=20,
+                        help="Amount less than the balance that we attempt to send")
+    return parser.parse_args()
+
+def read_file(path):
+    with open(path, "r") as file:
+        return file.read()
+
+def run_test(stub, amount, monitor_id, dest, max_seconds):
+    resp = stub.GetBalance(
+        mobilecoind_api_pb2.GetBalanceRequest(monitor_id=monitor_id))
+    starting_balance = resp.balance
+    print("Starting balance prior to transfer:", starting_balance)
+    tx_stats = {}
+    sync_start = time.time()
+    wait_for_accounts_sync(stub, [monitor_id], 3)
+    print("Time to sync:", time.time() - sync_start)
+    tx_resp = stub.SendPayment(
+        mobilecoind_api_pb2.SendPaymentRequest(
+            sender_monitor_id=monitor_id,
+            sender_subaddress=0,
+            outlay_list=[
+                mobilecoind_api_pb2.Outlay(
+                    value=amount,
+                    receiver=dest,
+                )
+            ],
+            fee=0,
+            tombstone=0,
+        ))
+
+    tx_stats[0] = {
+        'start': time.time(),
+        'time_delta': None,
+        'tombstone': tx_resp.sender_tx_receipt.tombstone,
+        'block_delta': None,
+        'status': TransferStatus.pending,
+        'receipt': tx_resp,
+    }
+    stats = poll(monitor_id, tx_stats, stub)
```

Comment:
Where is this defined?

Reply:
i am not sure, i will try to figure it out

Reply:
We should eventually have a pylint pre-commit hook available to make any python code more resilient in general. Given python is the Calvinball of languages, it's easy to footgun